pkg/didcomm/dispatcher/api.go

```diff
@@ -72,12 +72,18 @@ type Provider interface {
 // OutboundCreator method to create new outbound dispatcher service
 type OutboundCreator func(prov Provider) (Outbound, error)
 
+// ActionError structure for error handling
+type ActionError struct {
+	Err    error
+	Action *DIDCommAction
+}
+
 // DIDCommAction message type to pass events in go channels.
 type DIDCommAction struct {
 	// DIDComm message
 	Message DIDCommMsg
 	// Continue function to be called by the consumer for further processing the message.
-	Continue func()
+	Continue func(chan<- ActionError)
```

Comment:
It would be nice if we could avoid having to pas a nil argument when the caller doesn’t care about the error chan. I wonder if we should use a variadic pattern?

Reply:
Done!

Reply:
In my opinion, better to pass a clear argument, it's much easier to work with it. But options are not a bad approach, especially for callback (probably it will require one more option in the future)

Reply:
I like Rolson's snippet towards the end of [this comment](https://github.com/hyperledger/aries-framework-go/pull/393#issuecomment-539177203) better. It would be simpler from a user perspective if the error message has a `Retry()` function or something similar.

Reply:
It has `res.Action.Continue()` if you need retry it